src/ArmTemplates/Common/API/Clients/Abstractions/ApiClientBase.cs

```diff
@@ -0,0 +1,52 @@
+﻿// --------------------------------------------------------------------------
+//  <copyright file="ApiClientBase.cs" company="Microsoft">
+//      Copyright (c) Microsoft Corporation. All rights reserved.
+//  </copyright>
+// --------------------------------------------------------------------------
+
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Extensions;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Extractor.Utilities;
+using Microsoft.Extensions.Caching.Memory;
+
+namespace Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.API.Clients.Abstractions
+{
+    public abstract class ApiClientBase
+    {
+        public static string BaseUrl = "https://management.azure.com";
+
+        protected Authentication Auth = new Authentication();
+
+        static readonly IMemoryCache Cache = new MemoryCache(new MemoryCacheOptions());
+        static readonly HttpClient HttpClient = new HttpClient();
+
+        protected async Task<string> CallApiManagementAsync(string azToken, string requestUrl)
+        {
+            if (Cache.TryGetValue(requestUrl, out string cachedResponseBody))
+            {
+                return cachedResponseBody;
+            }
+
+            var request = new HttpRequestMessage(HttpMethod.Get, requestUrl);
+
```

Comment:
Nit: You seem to like empty lines a lot 😃. Can you at least group some things together instead of having an empty line between almost everything? E.g. this line and the empty line below `HttpResponseMessage response` don't need to exist.

Reply:
I do not like them either, it was just a moving of file :)
Fixed it as I could